### PR TITLE
move-cli: init at unstable-2022-08-28

### PR DESCRIPTION
**Pull request by macalinao**
###### Description of changes

Adds the Move CLI.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/move/move-cli/default.nix [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+, git
+}:
+
+let
+  common =
+    { buildFeatures ? [ ]
+    }:
+    rustPlatform.buildRustPackage rec {
+      pname = "move";
+      version = "f20499851934cd51f81b390954a292ca1bd419b8";
```
Comment (dit7ya) [inline pkgs/development/tools/move/move-cli/default.nix:18]: I think this should be in `unstable-YYYY-mm-dd` format - https://nixos.org/manual/nixpkgs/unstable/#sec-package-naming

Comment (macalinao) [reply pkgs/development/tools/move/move-cli/default.nix]: updated

**Comment by macalinao** [comment]
waiting on https://github.com/NixOS/nixpkgs/pull/188650 to be merged before I add tests for the prover

**Comment by macalinao** [comment]
also waiting on merge of https://github.com/NixOS/nixpkgs/pull/188987 for solc tests

**Comment by sifmelcara** [comment]
> also waiting on merge of https://github.com/NixOS/nixpkgs/pull/188987 for solc tests

Hi I got here from #188987 . I wonder why move needs older version of solc, is it because the tests are broken with newer solc or is there other reasons? I can't find this mentioned anywhere in https://github.com/move-language/move

**Comment by macalinao** [comment]
> > also waiting on merge of #188987 for solc tests
> 
> Hi I got here from #188987 . I wonder why move needs older version of solc, is it because the tests are broken with newer solc or is there other reasons? I can't find this mentioned anywhere in https://github.com/move-language/move

Yes the tests are broken with newer solc

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -24550,6 +24550,11 @@ with pkgs;
 
   moq = callPackage ../development/tools/moq { };
 
+  wrapWithMoveProverDeps = callPackage ../development/tools/move/wrapWithMoveProverDeps.nix { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:24553]:
```suggestion
  move-wrapWithMoveProverDeps = callPackage ../development/tools/move/wrapWithMoveProverDeps.nix { };
```

File: pkgs/development/tools/move/move-cli/default.nix [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+          "ci"
+          # Ignore Solidity tests since they require a specific version of Solc.
+          "--exclude"
+          "evm-exec-utils"
+          "--exclude"
+          "move-to-yul"
+        ] ++ (lib.optionals (!installProver) [
+          "--exclude"
+          "move-prover"
+        ]);
```
Comment (SuperSandro2000) [inline pkgs/development/tools/move/move-cli/default.nix:91]:
```suggestion
        ] ++ lib.optionals (!installProver) [
          "--exclude"
          "move-prover"
        ];
```

File: pkgs/development/tools/move/move-cli/default.nix [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+          Security
+          CoreFoundation
+          AppKit
+        ] ++ (lib.optionals stdenv.isAarch64 [ System ])))
+          ++ (lib.optionals installProver [
+          z3
+          icu
+          boogie
+          dotnet-sdk
+        ]);
```
Comment (SuperSandro2000) [inline pkgs/development/tools/move/move-cli/default.nix:63]:
```suggestion
        ] ++ lib.optionals stdenv.isAarch64 [ System ])
          ++ lib.optionals installProver [
          z3
          icu
          boogie
          dotnet-sdk
        ];
```

File: pkgs/development/tools/move/move-cli/default.nix [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+          owner = "move-language";
+          repo = "move";
+          rev = "bc57bf2df7aee3f639ec670af5c6ff5341d89a9d";
+          sha256 = "sha256-4BCwDTfdH8TTeF+zJyea3WsoU0YGgGSS1B3FxEZ6Ulk=";
+        };
+
+        cargoSha256 = "sha256-BTbChMtwSD5GIHLXUaEMlk3PMMh7jgR9yWk2W4J4El8=";
+
+        nativeBuildInputs = [ pkg-config ];
+        buildInputs = [ openssl zlib git ] ++ (lib.optionals stdenv.isDarwin ([
```
Comment (SuperSandro2000) [inline pkgs/development/tools/move/move-cli/default.nix:52]:
```suggestion
        buildInputs = [ openssl zlib git ]
          ++ lib.optionals stdenv.isDarwin ([
```

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (macalinao) [reply pkgs/top-level/all-packages.nix:24553]: Other CLIs should depend on this (eg Aptos/Sui)

Comment (SuperSandro2000) [reply pkgs/top-level/all-packages.nix:24553]: What do the programs have in common? is it maybe related to some framework?

**Pull request closed without merging**